ipfs-cluster-service/state.go

```diff
@@ -44,11 +42,62 @@ func upgrade() error {
 	}
 
 	// Reset with SnapshotSave
-	err = raft.SnapshotSave(consensusCfg, newState, clusterCfg.ID)
+	return raft.SnapshotSave(consensusCfg, newState, clusterCfg.ID)
+}
+
+func export(output string) error {
+	cfg, _, _, _, consensusCfg, _, _, _, _ := makeConfigs()
+
+	err := cfg.LoadJSONFromFile(configPath)
 	if err != nil {
 		return err
 	}
-	return nil
+
+	r, snapExists, err := raft.LastStateRaw(consensusCfg)
+	if err != nil {
+		return err
+	}
+	if !snapExists {
+		return errNoSnapshot
+	}
+
+	stateToExport := mapstate.NewMapState()
+	bs, err := ioutil.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	err = stateToExport.Unmarshal(bs)
```

Comment:
Calling Restore() here would do the same with the advantage that we'd upgrade any old state before exporting right ? We may wanna do that, unless I'm missing some downside